save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/CloneRepositoryController.kt

```diff
@@ -1,57 +1,40 @@
 package com.saveourtool.save.backend.controllers
 
 import com.saveourtool.save.backend.StringResponse
-import com.saveourtool.save.backend.configs.ConfigProperties
-import com.saveourtool.save.backend.service.ExecutionService
-import com.saveourtool.save.backend.service.ProjectService
+import com.saveourtool.save.backend.service.*
 import com.saveourtool.save.backend.storage.FileStorage
-import com.saveourtool.save.backend.utils.username
+import com.saveourtool.save.backend.storage.TestSuitesSourceSnapshotStorage
+import com.saveourtool.save.backend.utils.blockingToMono
 import com.saveourtool.save.domain.*
-import com.saveourtool.save.entities.Execution
-import com.saveourtool.save.entities.ExecutionRequest
-import com.saveourtool.save.entities.ExecutionRequestBase
-import com.saveourtool.save.entities.ExecutionRequestForStandardSuites
-import com.saveourtool.save.entities.Project
-import com.saveourtool.save.execution.ExecutionStatus
-import com.saveourtool.save.execution.ExecutionType
-import com.saveourtool.save.permission.Permission
+import com.saveourtool.save.entities.*
+import com.saveourtool.save.testsuite.TestSuitesSourceSnapshotKey
+import com.saveourtool.save.utils.orNotFound
 import com.saveourtool.save.v1
 
-import org.slf4j.LoggerFactory
-import org.springframework.boot.web.reactive.function.client.WebClientCustomizer
-import org.springframework.http.HttpStatus
 import org.springframework.http.ResponseEntity
 import org.springframework.security.core.Authentication
 import org.springframework.web.bind.annotation.PostMapping
 import org.springframework.web.bind.annotation.RequestMapping
 import org.springframework.web.bind.annotation.RequestPart
 import org.springframework.web.bind.annotation.RestController
-import org.springframework.web.reactive.function.client.WebClient
-import org.springframework.web.reactive.function.client.toEntity
 import reactor.core.publisher.Flux
 import reactor.core.publisher.Mono
+import reactor.kotlin.core.util.function.component1
+import reactor.kotlin.core.util.function.component2
 
 /**
  * Controller to save project
- *
- * @property projectService service to manage projects
- * @property configProperties configuration properties
  */
 @RestController
 @RequestMapping("/api")
 class CloneRepositoryController(
-    private val projectService: ProjectService,
-    private val executionService: ExecutionService,
+    private val testSuitesService: TestSuitesService,
+    private val testSuitesSourceService: TestSuitesSourceService,
+    private val testSuitesSourceSnapshotStorage: TestSuitesSourceSnapshotStorage,
     private val fileStorage: FileStorage,
-    private val configProperties: ConfigProperties,
-    jackson2WebClientCustomizer: WebClientCustomizer,
+    private val gitService: GitService,
+    private val runExecutionController: RunExecutionController,
```

Comment:
Never seen using controller inside another controller. Is it a good practice?

Reply:
I don't see a technical problem here, but it's not good for architecture (better avoid BL in controllers).
But it was added for transition period to leave old endpoints -- will be removed after implementing selection of IDs of TestSuite instead of current logic

Reply:
Oh ok